sdk/monitor/azquery/cloud_config.go

```diff
@@ -0,0 +1,35 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package azquery
+
+import "github.com/Azure/azure-sdk-for-go/sdk/azcore/cloud"
+
+const (
+	monitorQueryLogs    cloud.ServiceName = "azqueryLogs"
+	monitorQueryMetrics cloud.ServiceName = "azqueryMetrics"
+)
+
+func init() {
+	cloud.AzureChina.Services[monitorQueryLogs] = cloud.ServiceConfiguration{
+		Endpoint: "https://api.loganalytics.azure.cn",
+	}
+	cloud.AzureGovernment.Services[monitorQueryLogs] = cloud.ServiceConfiguration{
+		Endpoint: "https://api.loganalytics.us",
+	}
+	cloud.AzurePublic.Services[monitorQueryLogs] = cloud.ServiceConfiguration{
+		Endpoint: "https://api.loganalytics.io",
+	}
+	cloud.AzureChina.Services[monitorQueryMetrics] = cloud.ServiceConfiguration{
+		Endpoint: "https://management.chinacloudapi.cn/",
+	}
+	cloud.AzureGovernment.Services[monitorQueryMetrics] = cloud.ServiceConfiguration{
+		Endpoint: "https://management.usgovcloudapi.net/",
+	}
+	cloud.AzurePublic.Services[monitorQueryMetrics] = cloud.ServiceConfiguration{
+		Endpoint: "https://management.azure.com",
```

Comment:
MetricsClient talks to ARM, which makes me think we could simplify its constructor by calling `arm/runtime.NewPipeline()`. Come to think of it, I wonder whether we could simplify further by generating it as an ARM client 🤔

Reply:
leaving metrics client as is for now